frontend/test/unit/services/JsonRpcService.test.ts

```diff
@@ -0,0 +1,124 @@
+          leader_data: {
+            result: {
+              args: ['test'],
+              class: 'WizardOfCoin',
+              contract_state:
+                'gASVNwAAAAAAAACMCF9fbWFpbl9flIwMV2l6YXJkT2ZDb2lulJOUKYGUfZSMCWhhdmVfY29pbpSMBFRydWWUc2Iu',
+              eq_outputs: {
+                leader: {
+                  '0': '{\n"reasoning": "I cannot give you the coin because it holds powerful magic that must not fall into the wrong hands.",\n"give_coin": False\n}'
+                }
+              },
+              gas_used: 0,
+              method: 'ask_for_coin',
+              mode: 'leader',
+              node_config: {
+                address: '0xB1971EAfB15Fd2a04afAd55A7d5eF9940c0dd464',
+                config: {},
+                id: 1,
+                model: 'gpt-3.5-turbo',
+                provider: 'openai',
+                stake: 6.9066502309882605,
+                type: 'validator',
+                updated_at: '05/27/2024, 19:50:36'
+              }
+            },
+            vote: 'agree'
+          }
+        }
+      },
+      message: '',
+      status: 'success'
+    }
+    it('should call rpc client', async () => {
+      vi.spyOn(rpcClient, 'call').mockImplementationOnce(() => Promise.resolve(mockResult))
+      const input = {
+        userAccount: '0xFEaedeC4c6549236EaF49C1F7c5cf860FD2C3fcB',
+        contractAddress: '0x58FaA28cbAA1b52F8Ec8D3c6FFCE6f1AaF8bEEB1',
+        method: 'WizardOfCoin.ask_for_coin',
+        params: ['Give me the coin']
+      }
+      await jsonRpcService.callContractFunction(input)
+      expect(rpcClient.call).toHaveBeenCalledTimes(1)
+      expect(rpcClient.call).toHaveBeenCalledWith(
+        {
+          method: 'call_contract_function',
+          params: [input.userAccount, input.contractAddress, input.method, [...input.params]]
+        }
+      )
+    })
+
+    it('should call contract function and return result', async () => {
```

Comment:
@bradleySuira same comment as in line 43

Reply:
fixed